driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/pageobjects/internal/action/GuiElementCheckFieldAction.java

```diff
@@ -100,6 +124,9 @@ private void pCheckField(UiElementBase guiElement, Check check) {
         if (prevTimeout >= 0) {
             overrides.setTimeout(prevTimeout);
         }
+        if (check.optional() || check.collect()) {
```

Comment:
No strong preference either way, but heads-up that it might be better to keep the code for `previousAssertionImpl` together instead of splitting it with a largish blob of general code inbetween. Similarly, `prevTimeout` has last been used all the way above handling `previousAssertionImpl`.
... But dunno whether locality trumps process definition in this case.

Reply:
I cannot move that, because the UiElement check logic is done some lines above.